resolvewithplus.js

```diff
@@ -191,6 +203,91 @@ export default (o => {
     return o.getasfilesync(temppath, opts) || o.getasdirsync(temppath, opts);
   };
 
+  o.getasesmimportpathfrompjson = (targetpath, specifier, pjson) => {
```

Comment:
haha ok

Reply:
this was renamed to "esmparseimportpkg" if you prefer a different name it could be changed and readability improvements could be made in a follow up PR